common/config/config_parser.go

```diff
@@ -278,6 +276,10 @@ func IsSet(key string) bool {
 	return viper.IsSet(key)
 }
 
+func IsCLISet(key string) bool {
```

Comment:
So now we are putting this extra condition check in all components. Can we instead call this implicitly inside IsSet() itself so that component need to call just one method. I understand that cli and config param name differs but those clis are also mapped to the component config so there might be a way to do this.

Reply:
only issue here is the cli flag and config flag name differs so we need to see how a single call can resolve this

Reply:
I've addressed this